components/support/crypto/nss_sys/src/lib.rs

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#![allow(non_camel_case_types, non_upper_case_globals, non_snake_case)]
+
+mod types;
+
+use std::os::raw::{c_char, c_uchar};
+pub use types::nspr::*;
+pub use types::nss3::*;
+
+// Code adapted from https://stackoverflow.com/a/35591693. I am not this kind of smart.
+macro_rules! nss_exports {
+    () => {};
+    (
+        unsafe fn $i:ident($($arg:ident: $argty:ty),*) -> $ret:ty;
+        $($tail:tt)*
+    ) => {
+        #[cfg(not(target_os = "ios"))]
+        lazy_static::lazy_static! {
+            pub static ref $i: libloading::Symbol<'static, unsafe extern fn($($arg: $argty),*) -> $ret> = {
+                unsafe {
+                    LIBNSS3.get(stringify!($i).as_bytes()).expect(stringify!(Could not get $i handle))
+                }
+            };
+        }
+        #[cfg(target_os = "ios")]
+        #[link(name = "nss3")]
+        #[link(name = "nspr4")]
+        extern "C" {
+            pub fn $i($($arg: $argty),*) -> $ret;
+        }
+        nss_exports! { $($tail)* }
+    };
+    // Support for functions that don't return can be added by copy-pasting the above code and removing -> ref:ty.
+}
+
+#[cfg(not(target_os = "ios"))]
+lazy_static::lazy_static! {
+    // Lib handle.
+    static ref LIBNSS3: libloading::Library = {
+        #[cfg(any(target_os = "macos", target_os = "ios"))]
+        const LIB_NAME: &str = "libnss3.dylib";
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        const LIB_NAME: &str = "libnss3.so";
+        #[cfg(target_os = "windows")]
+        const LIB_NAME: &str = "nss3.dll";
+        libloading::Library::new(LIB_NAME).expect("Cannot load libnss3.")
```

Comment:
As mentioned in meeting just now, if there's a cheap way to assert that we've loaded a compatible version of NSS here, we should do so in order to guard against GeckoView shipping new versions without us knowing.

Reply:
NSS has `NSS_VersionCheck`, pretty good!